clickhouse/sg.tf

```diff
@@ -32,6 +42,49 @@ resource "aws_security_group" "clickhouse_cluster" {
   name        = "clickhouse_cluster-sg"
   description = "Security group for the ClickHouse cluster"
   vpc_id      = module.vpc.vpc_id
+
+  # Prometheus metrics
+  ingress {
+    description = "Prometheus metrics"
+    from_port   = var.prometheus_port
+    to_port     = var.prometheus_port
+    protocol    = "tcp"
+    self        = true
+  }
+
+  # Dynamic block for encrypted traffic
+  dynamic "ingress" {
+    for_each = var.enable_encryption ? [1] : []
+    content {
+      description = "Secure ClickHouse traffic"
+      from_port   = var.tcp_port_secure
+      to_port     = var.tcp_port_secure
+      protocol    = "tcp"
+      self        = true
+    }
+  }
+
+  # Dynamic block for unencrypted traffic
+  dynamic "ingress" {
+    for_each = var.enable_encryption ? [] : [1]
+    content {
+      description = "ClickHouse traffic"
+      from_port   = var.tcp_port
+      to_port     = var.tcp_port
+      protocol    = "tcp"
+      self        = true
+    }
+  }
+
+  # Allow all outbound traffic
+  egress {
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
```

Comment:
I think we can get rid of [this](https://github.com/cloudquery/terraform-cloudquery-modules/pull/14/files#diff-3d2d5996ab054f19e957f3dba57e1a1e61655aa027eb53a97c31f7467e19104fR91-R98) if using an inline rule